lib/foreman/renderer/scope/macros/base.rb

```diff
@@ -49,10 +99,28 @@ def pxe_kernel_options
             ''
           end
 
+          apipie :method, 'Generates a shell command to store the given text into a file' do
+            desc "This is useful if some multline string needs to be saved somewhere on the harddisk. This
+              is typically used in provisioning or job templates, e.g. when puppet configuration file is
+              generated based on host configuration and stored for puppet agent."
+            required :filename, String, desc: 'the file path to store the content to'
+            required :content, String, desc: 'content to be stored'
+            returns String, desc: 'String representing the shell command'
+            example "save_to_file(\"hello\\nworld\\n\", '/etc/motd') # => 'cat << EOF > /etc/motd\\nhello\\nworld\\nEOF'"
+          end
           def save_to_file(filename, content)
             "cat << EOF > #{filename}\n#{content}EOF"
```

Comment:
Not part of the patch, but I'd probably avoid the generic "EOF" which people use a lot.